### PR TITLE
Update `smithay`

### PR DESCRIPTION
**Pull request by ids1024**

### CODE REVIEW

File: src/wayland/handlers/input_method.rs [hunk 1]
```diff
@@ -32,6 +32,8 @@ impl InputMethodHandler for State {
             .map(|e| e.geometry())
             .unwrap_or_default()
     }
+
+    fn popup_repositioned(&mut self, _: PopupSurface) {}
```
Comment (Drakulix) [inline src/wayland/handlers/input_method.rs:36]: We probably should handle this similarly to how niri does it, no?

Comment (ids1024) [reply src/wayland/handlers/input_method.rs:36]: Probably. I haven't looked at how niri implements it.

**Comment by Vixea** [comment]
Not to rush this or anything but can the git commit be updated to include the pointer fix while I don't know for sure I have a feeling it might fix a recent regression with pointers in fullscreen applications

**Comment by Vixea** [comment]
Umm now xwayland is completely broken. Probably do to this?
https://github.com/Smithay/smithay/commit/8773a74210214556b028ebce3f9c5e57582851e6
Though not sure if it's a smithay problem or cosmic problem

**Comment by ids1024** [comment]
Hm, at least in some quick testing XWayland works fine for me. (`xterm` and `gimp` start and generally seem to work.)

**Comment by Vixea** [comment]
Have you checked steam it's the only application I really opened because of the test case

**Comment by Vixea** [comment]
Yea it's saying display is not set even on xterm

**Comment by ids1024** [comment]
Seems like a timing thing. When I start with cosmic-session, I can't start xterm from the launcher. But if I open a terminal with

**Comment by Vixea** [comment]
Huh that's funny because I tried the terminal too, but it didn't work

File: src/wayland/handlers/input_method.rs (hunk 1 above)
Comment (Drakulix) [reply src/wayland/handlers/input_method.rs:36]: https://github.com/YaLTeR/niri/pull/295

**Comment by Drakulix** [comment]
Added some changes to lazy initialize the rest of the session, once we got the DISPLAY from xwayland. Seems to work well in my testing. @Vixea does this fix your issues?

**Comment by ryanabx** [comment]
This will be necessary (and probably the final hurdle) to getting cosmic-comp on Fedora Rawhide :partying_face:

**Comment by ids1024** [comment]
Updated to fix minor merge conflict.

**Comment by Vixea** [comment]
I'm sadly unable to test btw

**Comment by Drakulix** [comment]
Since Xwayland's DISPLAY seems to propagate correctly now and I don't see other regressions, I am going ahead and merge this. Everything else can be handled later.

**Pull request merged**